### ISSUE TITLE
Nunjucks: async shortcode inside paired shortcode

### ISSUE THREAD

**Issue by solemone** [11ty/eleventy#1053]
Hi Community!

**Describe the bug**
I want to use an async shortcode inside a paired shortcode. Unfortunately there is no content from the async shortcode (child) if it is inside the paired shortcode (parent). Standalone the async shortcode works. 

**To Reproduce**
I abstracted my code to this:

**Shortcodes (inside template file)**
```
{% parentPairedShortcode %}
  {% asyncChildShortcode %}
{% endparentPairedShortcode %}
```

**Config (.eleventy.js)**
```javascript
async function getSomething() {
  const promise = new Promise((resolve, reject) => {
    setTimeout(() => resolve("done!"), 1000)
  });
  const result = await promise;
  return result;
}

eleventyConfig.addNunjucksAsyncShortcode("asyncChildShortcode", async function() {
  const something = await getSomething();
  return something;
});

eleventyConfig.addPairedNunjucksAsyncShortcode("parentPairedShortcode", async function(content) {
  return content;
});
```

**Expected behavior**
Get the content of the async shortcode (child) paired shortcode back. I tried that with liquid shortcodes and it works. Did I do something wrong? Or did I forget something?

**Environment:**
- macOS 10.15.4 
- Eleventy v0.10.0

Please let me know if more information is needed. Thank you in advance.

**Comment by S-TheCoder** [comment]
@solemone , can you share the sample code or config how you were able to use async short code in Nunjucks, I have been trying to use that but with no success.

**Comment by solemone** [comment]
Hi @S-TheCoder,
a single async shortcode works fine. If you use {% asyncChildShortcode %} from my example above as standalone shortcode it should work. But it still does not work inside a async paired shortcode. For this I switched to liquid template language: https://www.11ty.dev/docs/languages/liquid/

**Comment by node-ex** [comment]
Recently I have encountered the same problem, when I tried to to use a **style** shortcode from Twelvety template (https://github.com/gregives/Twelvety) with Nunjucks.

Here is a simple showcase of a problem in CodeSandbox: https://codesandbox.io/s/eleventy-nunjucks-async-nunjucks-shortcode-problem-s2mxu?file=/src/index.njk

It looks like there is no such problem with Liquid template language, but I like working with Nunjucks more and would be grateful for an example of an appropriate usage or fix.

**Comment by zachleat** [comment]
Not sure if y’all are following but some discussion on this one over at https://github.com/11ty/eleventy/pull/1749

**Comment by zachleat** [comment]
This will ship with `1.0.0-beta.9` thanks to the very lovely #1749

**Comment by zachleat** [comment]
For future me, note that the fix here was included in 1.0 stable but the async-everything approach was reverted in #2140.
